# Laboratory (old)

<div id="toc"></div>

In [1]:
%%javascript
$.getScript('https://kmahelona.github.io/ipython_notebook_goodies/ipython_notebook_toc.js')
// https://github.com/kmahelona/ipython_notebook_goodies          ...interactive table of contents

<IPython.core.display.Javascript object>

In [3]:
%cd "d:\moje_dokumenty\Desktop\diplomka\model\data\hdf\" 

d:\moje_dokumenty\Desktop\diplomka\model\data\hdf


In [4]:
import model_tools 

In [5]:
Sim = model_tools.Simulation
experiment = model_tools.experiment
random = model_tools.random
np = model_tools.np
exp_grid = model_tools.exp_grid
make_matrix = model_tools.make_matrix
make_grid = model_tools.make_grid

In [6]:
def set_base_pars(ns = None):
    """Sets all experiment-dependend parameters to base levels."""
    global model_tools
    if ns is not None:         # leave unchanged by default
        model_tools.ns = ns
        model_tools.ms = [int(11*n/100) for n in ns]
    model_tools.global_bank = False
    model_tools.transport_firms = False                                        ##################################
    model_tools.no_countries = len(model_tools.ns)
    
##    model_tools.ns = [5000,5000]    # no HSHs
##    model_tools.ms = [550,550]      # no firms
    
    model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
    model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
    
    model_tools.trade_costs = np.array([[0.05]*model_tools.no_countries]*model_tools.no_countries)                         ## tau
    np.fill_diagonal(model_tools.trade_costs, [0]*model_tools.no_countries)                              ## keep as np array 
    
    model_tools.free_market = True                 # all quotas are 1?
    model_tools.trade_quotas = np.array([[1]*model_tools.no_countries]*model_tools.no_countries)                                     
    np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
    model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 
    
    model_tools.pegged_rates = [-1]*model_tools.no_countries     # which currencies are pegged together, -1 if to none
    model_tools.trade_sensitivity = 0.1              # sensitivity in exchange factors updating    ## gamma
    model_tools.normal_variance = 0.002
    
    if model_tools.no_countries == 2:
        model_tools.providers = 15                 # market selection - no of firms considered by each hsh
    elif model_tools.no_countries == 6:
        model_tools.providers = 30                
    else:
        model_tools.providers = 10
        
    model_tools.p = 0.7

In [7]:
#rand_seeds = [[24, 24]] + [[random.randrange(10**5), random.randrange(10**5)] for i in range(30) ]   # for 31 runs

In [8]:
rand_seeds = [[24, 24],
 [57705, 84477],
 [21235, 79339],
 [72193, 63946],
 [41209, 89019],
 [78694, 83426],
 [71793, 98870],
 [52518, 26074],
 [96107, 49709],
 [26813, 89835],
 [42384, 58898],
 [23077, 29772],
 [75905, 1768],
 [54715, 58905],
 [39040, 79913],
 [79944, 42114],
 [76822, 52962],
 [14277, 8719],
 [71510, 32894],
 [94253, 43188],
 [16195, 43158],
 [10706, 35992],
 [92209, 64592],
 [4414, 36487],
 [37706, 44597],
 [5180, 75662],
 [71206, 9908],
 [49354, 2217],
 [46671, 39293],
 [91634, 58719],
 [51505, 84003]]

In [8]:
model_tools.parameters()

{'period': 'current period = -1',
 'no_countries': 'number of countries = 2',
 'global_bank': 'if there is only 1 global bank = False',
 'ns': 'number of HSHs = [5000, 5000]',
 'ms': 'number of firms = [550, 550]',
 'wage_resist': 'hsh - wage resistance in unemployment = [12, 12]',
 'wage_flex': 'hsh - wage flexibility = [0.05, 0.05]',
 'prop_to_save_O': 'hsh - optimistic saving target = 0.15',
 'prop_to_save_P': 'hsh - pessimistic saving target = 0.2',
 'cons_of_excess_savings': 'hsh - rate of cons. of excess savings = 0.5',
 'l_providers': 'hsh - number of firms to be considered on labor market = 10',
 'providers': 'hsh - number of firms to be considered on goods market = 15',
 'K': 'f - number of machines = 10',
 'productivity': 'f - machine productivity = 100',
 'production_time': 'f - length of production = 8',
 'inv_normal': 'f - periods of full prod. for targeted inventories = 2',
 'price_flex': 'f - adjustment parameter = 0.05',
 'targeted_vacancies': 'f - targeted rate of vaca

# 1 country

In [37]:
model_tools.ns = [5000]    # no HSHs
model_tools.ms = [550]      # no firms
set_base_pars()


### baseline

In [9]:
model_tools.wage_resist = [12]
model_tools.wage_flex = [0.05]

In [11]:
for seeds in rand_seeds[:7]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1500)                     # 125 years
    aa.finalize_data()
    aa.export_data(note = "Base")

C:\Users\am\model_tools.py:1631: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1571: RuntimeWarning: overflow encountered in long_scalars
  res = [sum(country[shift+i:(shift+i+perioda)]) for i in range(0,end,perioda)]         # take only full years


In [10]:
experiment(rand_seeds[:7], 1500, "Base_corrected")

C:\Users\am\model_tools.py:1539: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1525: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1539: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1767: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_sentiment', 'hsh_income', 'savings_T',
       'budget', 'cons', 'cons_value', 'forced_savings', 'hsh_deposits',
       'prices', 'offered_wages', 'production', 'production_value',
       'labor_demand', 'labor', 'job_offers', 'vacancies', 'inventories',
       'sales', 'sales_value', 'sales_costs',

### alternative 

In [11]:
model_tools.wage_resist = [8]
model_tools.wage_flex = [0.08]

In [13]:
for seeds in rand_seeds[:7]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1500)                     # 125 years
    aa.finalize_data()
    aa.export_data(note = "Alt")

In [12]:
experiment(rand_seeds[:7], 1500, "Alt_corrected")

### extreme alternative

In [13]:
model_tools.wage_resist = [8]
model_tools.wage_flex = [0.25]

In [15]:
for seeds in rand_seeds[:7]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(840)                     # 80 years is enough, usually bank goes bankrupt around year 25
    aa.finalize_data()
    aa.export_data(note = "Extr_alt")

345:  Bank 0_b_0 is bankrupt!
323:  Bank 0_b_0 is bankrupt!
311:  Bank 0_b_0 is bankrupt!
265:  Bank 0_b_0 is bankrupt!
280:  Bank 0_b_0 is bankrupt!
598:  Bank 0_b_0 is bankrupt!
296:  Bank 0_b_0 is bankrupt!


messages from extreme_alternative (rand_seeds[:7]):

345:  Bank 0_b_0 is bankrupt!

323:  Bank 0_b_0 is bankrupt!

311:  Bank 0_b_0 is bankrupt!

265:  Bank 0_b_0 is bankrupt!

280:  Bank 0_b_0 is bankrupt!

598:  Bank 0_b_0 is bankrupt!

296:  Bank 0_b_0 is bankrupt!

In [14]:
experiment(rand_seeds[:7], 1500, "alt_Extr_corrected")

649:  Bank 0_b_0 is bankrupt!
1142:  Bank 0_b_0 is bankrupt!
255:  Bank 0_b_0 is bankrupt!
281:  Bank 0_b_0 is bankrupt!
348:  Bank 0_b_0 is bankrupt!
258:  Bank 0_b_0 is bankrupt!
283:  Bank 0_b_0 is bankrupt!


messages from extreme_alternative (rand_seeds[:7]):

649:  Bank 0_b_0 is bankrupt!

1142:  Bank 0_b_0 is bankrupt!

255:  Bank 0_b_0 is bankrupt!

281:  Bank 0_b_0 is bankrupt!

348:  Bank 0_b_0 is bankrupt!

258:  Bank 0_b_0 is bankrupt!

283:  Bank 0_b_0 is bankrupt!

# 2 large countries

In [32]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()


## 2 closed economies (for comfortable comparisons)

In [12]:
model_tools.free_market = False
model_tools.trade_quotas = make_matrix(0)

In [13]:
experiment(rand_seeds[7:10], 1200, note = "closed_base_bb", csv = False)

C:\Users\am\model_tools.py:1556: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1542: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1556: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1789: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_sentiment', 'hsh_income', 'savings_T',
       'budget', 'cons', 'cons_value', 'forced_savings', 'hsh_deposits',
       'prices', 'offered_wages', 'production', 'production_value',
       'labor_demand', 'labor', 'job_offers', 'vacancies', 'inventories',
       'sales', 'sales_value', 'sales_costs',

In [14]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08, 0.25]             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
experiment(rand_seeds[7:10], 1200, note = "closed_alts_bb", csv = False)

507:  Bank 1_b_0 is bankrupt!
460:  Bank 1_b_0 is bankrupt!
907:  Bank 1_b_0 is bankrupt!


### baseline bb

Free trade, baseline wage and exchange rates flexibility.

In [ ]:
set_base_pars()

In [40]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

#model_tools.trade_quotas = np.array([[1]*model_tools.no_countries]*model_tools.no_countries) # free trade                                    


In [18]:
for seeds in rand_seeds[7:14]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "Base")

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


With corrected sales value of firms selling in foreign markets:

In [41]:
experiment(rand_seeds[7:14], 1200, "Base_corrected")

### alternative 1 bb

Free trade, baseline exchange rates flexibility, alternative wage flexibility in both countries.

In [42]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [20]:
for seeds in rand_seeds[7:14]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "Alt1")

With corrected sales value of firms selling in foreign markets:

In [43]:
experiment(rand_seeds[7:14], 1200, "Alt1_corrected")

### alternative 2 bb

Free trade and baseline exchange rates flexibility, but 1 country has alternative wage flexibility, the other baseline one.

In [44]:
model_tools.wage_resist = [12, 8]          # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.08]         # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [22]:
for seeds in rand_seeds[7:14]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "Alt2")

With corrected sales value of firms selling in foreign markets:

In [45]:
experiment(rand_seeds[7:14], 1200, "Alt2_corrected")

### extreme alternative 1 bb

Free trade, baseline exchange rates flexibility, extreme alternative wage flexibility in both countries.

* Collapse happens before period 200, 350 periods should be enough.

In [46]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [24]:
for seeds in rand_seeds[7:14]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years          ## try fewer runs, probably enough
    aa.finalize_data()
    aa.export_data(note = "Extr_alt1")

308:  Bank 1_b_0 is bankrupt!
311:  Bank 0_b_0 is bankrupt!
298:  Bank 1_b_0 is bankrupt!
308:  Bank 0_b_0 is bankrupt!
273:  Bank 0_b_0 is bankrupt!
286:  Bank 1_b_0 is bankrupt!
245:  Bank 1_b_0 is bankrupt!
246:  Bank 0_b_0 is bankrupt!
385:  Bank 0_b_0 is bankrupt!
388:  Bank 1_b_0 is bankrupt!
298:  Bank 0_b_0 is bankrupt!
301:  Bank 1_b_0 is bankrupt!
382:  Bank 0_b_0 is bankrupt!
396:  Bank 1_b_0 is bankrupt!


messages from incorrect extreme alternative 1 (rand_seeds[7:14]):

308:  Bank 1_b_0 is bankrupt!

311:  Bank 0_b_0 is bankrupt!

298:  Bank 1_b_0 is bankrupt!

308:  Bank 0_b_0 is bankrupt!

273:  Bank 0_b_0 is bankrupt!

286:  Bank 1_b_0 is bankrupt!

245:  Bank 1_b_0 is bankrupt!

246:  Bank 0_b_0 is bankrupt!

385:  Bank 0_b_0 is bankrupt!

388:  Bank 1_b_0 is bankrupt!

298:  Bank 0_b_0 is bankrupt!

301:  Bank 1_b_0 is bankrupt!

382:  Bank 0_b_0 is bankrupt!

396:  Bank 1_b_0 is bankrupt!

With corrected sales value of firms selling in foreign markets:

In [47]:
experiment(rand_seeds[7:14], 1200, "Extr_alt1_corrected")

153:  Bank 0_b_0 is bankrupt!
157:  Bank 1_b_0 is bankrupt!
163:  Bank 0_b_0 is bankrupt!
164:  Bank 1_b_0 is bankrupt!
151:  Bank 1_b_0 is bankrupt!
153:  Bank 0_b_0 is bankrupt!
161:  Bank 1_b_0 is bankrupt!
164:  Bank 0_b_0 is bankrupt!
171:  Bank 1_b_0 is bankrupt!
172:  Bank 0_b_0 is bankrupt!
160:  Bank 1_b_0 is bankrupt!
162:  Bank 0_b_0 is bankrupt!
160:  Bank 1_b_0 is bankrupt!
163:  Bank 0_b_0 is bankrupt!


messages from corrected extreme alternative 1 (rand_seeds[7:14]):

153:  Bank 0_b_0 is bankrupt!

157:  Bank 1_b_0 is bankrupt!

163:  Bank 0_b_0 is bankrupt!

164:  Bank 1_b_0 is bankrupt!

151:  Bank 1_b_0 is bankrupt!

153:  Bank 0_b_0 is bankrupt!

161:  Bank 1_b_0 is bankrupt!

164:  Bank 0_b_0 is bankrupt!

171:  Bank 1_b_0 is bankrupt!

172:  Bank 0_b_0 is bankrupt!

160:  Bank 1_b_0 is bankrupt!

162:  Bank 0_b_0 is bankrupt!

160:  Bank 1_b_0 is bankrupt!

163:  Bank 0_b_0 is bankrupt!

### extreme alternative 2 bb

Free trade and baseline exchange rates flexibility, but 1 country has extreme alternative wage flexibility, the other baseline one.

In [48]:
model_tools.wage_resist = [12, 8]          # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.25]         # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [26]:
for seeds in rand_seeds[7:14]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                   # 100 years            ## try fewer runs, probably enough ?
    aa.finalize_data()
    aa.export_data(note = "Extr_alt2")

With corrected sales value of firms selling in foreign markets:

In [49]:
experiment(rand_seeds[7:14], 1200, "Extr_alt2_corrected")

## symmetrical BB alternatives

In [35]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [36]:
experiment(rand_seeds[7:10], 1200, "Alt_sym_BB", csv = False)

In [37]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
experiment(rand_seeds[7:10], 600, "Alt_extr_sym_BB", csv = False)

182:  Bank 1_b_0 is bankrupt!
183:  Bank 0_b_0 is bankrupt!
158:  Bank 1_b_0 is bankrupt!
159:  Bank 0_b_0 is bankrupt!
148:  Bank 0_b_0 is bankrupt!
150:  Bank 1_b_0 is bankrupt!


# 1 big 1 small country

Size of countries is changed. Test baseline and then only heterogenous wage flexibilities, first in the small country, then in the large one.

In [27]:
model_tools.ns = [8000,2000]    # no HSHs
model_tools.ms = [880,220]      # no firms
set_base_pars()

## BS closed economies (for comfortable comparisons)

In [16]:
model_tools.free_market = False
model_tools.trade_quotas = make_matrix(0)

In [17]:
experiment(rand_seeds[14:17], 1200, note = "closed_base_bs", csv = False)

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


In [18]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08, 0.08]             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
experiment(rand_seeds[14:17], 1200, note = "closed_alt_bs", csv = False)

In [19]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25, 0.25]             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
experiment(rand_seeds[14:17], 1200, note = "closed_extr_alt_bs", csv = False)

406:  Bank 1_b_0 is bankrupt!
227:  Bank 1_b_0 is bankrupt!
270:  Bank 1_b_0 is bankrupt!
478:  Bank 0_b_0 is bankrupt!


### baseline bs

Free trade, baseline wage and exchange rates flexibility.

In [ ]:
set_base_pars()

In [29]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

#model_tools.trade_quotas = np.array([[1]*model_tools.no_countries]*model_tools.no_countries) # free trade                                    


In [29]:
for seeds in rand_seeds[14:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Base")

With corrected sales value of firms selling in foreign markets:

In [30]:
experiment(rand_seeds[14:21], 1200, "Base_big_small_corrected")

### alternative 1 bs

Free trade, baseline exchange rates flexibility, alternative wage flexibility in small country.

In [31]:
model_tools.wage_resist = [12, 8]             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.08]             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [31]:
for seeds in rand_seeds[14:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Alt1")

With corrected sales value of firms selling in foreign markets:

In [32]:
experiment(rand_seeds[14:21], 1200, "Alt1_cbig_small_orrected")

### alternative 1b bs

Free trade and baseline exchange rates flexibility, big country has alternative wage flexibility, small one has baseline one.

In [33]:
model_tools.wage_resist = [8, 12]          # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08, 0.05]         # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [33]:
for seeds in rand_seeds[14:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Alt1b")

With corrected sales value of firms selling in foreign markets:

In [34]:
experiment(rand_seeds[14:21], 1200, "Alt1b_big_small_corrected")

809:  Bank 1_b_0 is bankrupt!
888:  Bank 1_b_0 is bankrupt!
779:  Bank 1_b_0 is bankrupt!
816:  Bank 1_b_0 is bankrupt!
852:  Bank 1_b_0 is bankrupt!
922:  Bank 1_b_0 is bankrupt!
914:  Bank 1_b_0 is bankrupt!


corrected:

809:  Bank 1_b_0 is bankrupt!

888:  Bank 1_b_0 is bankrupt!

779:  Bank 1_b_0 is bankrupt!

816:  Bank 1_b_0 is bankrupt!

852:  Bank 1_b_0 is bankrupt!

922:  Bank 1_b_0 is bankrupt!

914:  Bank 1_b_0 is bankrupt!

### extreme alternative (2) bs

Free trade, baseline exchange rates flexibility, extreme alternative wage flexibility in small country.



In [35]:
model_tools.wage_resist = [12, 8]              # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.25]          # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [35]:
for seeds in rand_seeds[14:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Alt2")

With corrected sales value of firms selling in foreign markets:

In [36]:
experiment(rand_seeds[14:21], 1200, "Alt2_big_small_corrected")

### extreme alternative (2b) bs

Free trade, baseline exchange rates flexibility, extreme alternative wage flexibility in big country.



In [37]:
model_tools.wage_resist = [8, 12]              # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25, 0.05]          # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [ ]:
for seeds in rand_seeds[14:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Alt2b")

804:  Bank 1_b_0 is bankrupt!
532:  Bank 1_b_0 is bankrupt!
470:  Bank 1_b_0 is bankrupt!


In [8]:
# computer shut down, finishing previous
for seeds in rand_seeds[17:21]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "big_small_Alt2b_completing")

473:  Bank 1_b_0 is bankrupt!


C:\Users\am\model_tools.py:1646: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


522:  Bank 1_b_0 is bankrupt!
450:  Bank 1_b_0 is bankrupt!
1171:  Bank 1_b_0 is bankrupt!


C:\Users\am\model_tools.py:1586: RuntimeWarning: overflow encountered in long_scalars
  res = [sum(country[shift+i:(shift+i+perioda)]) for i in range(0,end,perioda)]         # take only full years


extreme alternative 2b messages (rand_seeds[14:21]):

804:  Bank 1_b_0 is bankrupt!

532:  Bank 1_b_0 is bankrupt!

470:  Bank 1_b_0 is bankrupt!

473:  Bank 1_b_0 is bankrupt!

522:  Bank 1_b_0 is bankrupt!

450:  Bank 1_b_0 is bankrupt!

1171:  Bank 1_b_0 is bankrupt!

With corrected sales value of firms selling in foreign markets:

In [38]:
experiment(rand_seeds[14:21], 1200, "Alt2b_big_small_corrected")

501:  Bank 1_b_0 is bankrupt!
513:  Bank 1_b_0 is bankrupt!
543:  Bank 1_b_0 is bankrupt!
352:  Bank 1_b_0 is bankrupt!
293:  Bank 1_b_0 is bankrupt!
470:  Bank 1_b_0 is bankrupt!
504:  Bank 1_b_0 is bankrupt!


501:  Bank 1_b_0 is bankrupt!

513:  Bank 1_b_0 is bankrupt!

543:  Bank 1_b_0 is bankrupt!

352:  Bank 1_b_0 is bankrupt!

293:  Bank 1_b_0 is bankrupt!

470:  Bank 1_b_0 is bankrupt!

504:  Bank 1_b_0 is bankrupt!

## symmetrical BS alternatives

In [28]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

In [30]:
experiment(rand_seeds[14:17], 1200, "Alt_sym_big_small_corrected", csv = False)

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
C:\Users\am\model_tools.py:1556: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1542: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1556: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1789: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_senti

In [31]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
experiment(rand_seeds[14:17], 1200, "Alt_extr_sym_big_small_corrected", csv = False)

183:  Bank 1_b_0 is bankrupt!
187:  Bank 0_b_0 is bankrupt!
157:  Bank 1_b_0 is bankrupt!
161:  Bank 0_b_0 is bankrupt!
157:  Bank 1_b_0 is bankrupt!
165:  Bank 0_b_0 is bankrupt!


# Experiments with exchange rate sensitivities

## 2 large countries experiments

In [30]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()


### baseline experiments

In [31]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

model_tools.free_market = True                                  


#### pegged rates

In [27]:
model_tools.pegged_rates = [-1,0]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.1              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [28]:
#experiment(rand_seeds[21:24], 1200, "exch_1")

In [9]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_1")

C:\Users\am\model_tools.py:1655: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1595: RuntimeWarning: overflow encountered in long_scalars
  res = [sum(country[shift+i:(shift+i+perioda)]) for i in range(0,end,perioda)]         # take only full years
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


With corrected sales value of firms selling in foreign markets:

In [29]:
experiment(rand_seeds[21:24], 1200, "exch_0_corrected")

#### 0.00 sensitivity
only random disturbances in exchange rates

In [30]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.00              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [31]:
experiment(rand_seeds[21:24], 1200, "exch_1_corrected")

#### 0.05 sensitivity

In [32]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.05              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_2")

In [11]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_2")

With corrected sales value of firms selling in foreign markets:

In [33]:
experiment(rand_seeds[21:24], 1200, "exch_2_corrected")

#### 0.15 sensitivity

In [34]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.15              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_3")

In [13]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_3")

With corrected sales value of firms selling in foreign markets:

In [35]:
experiment(rand_seeds[21:24], 1200, "exch_3_corrected") 

#### 0.2 sensitivity

In [36]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.2              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_4")

In [15]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_4")

With corrected sales value of firms selling in foreign markets:

In [37]:
experiment(rand_seeds[21:24], 1200, "exch_4_corrected")

#### 0.4 sensitivity

In [38]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.4              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_5")

In [17]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_5")

With corrected sales value of firms selling in foreign markets:

In [39]:
experiment(rand_seeds[21:24], 1200, "exch_5_corrected")

#### 0.7 sensitivity

In [40]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 0.7              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_6")

In [19]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_6")

With corrected sales value of firms selling in foreign markets:

In [41]:
experiment(rand_seeds[21:24], 1200, "exch_6_corrected")

985:  Bank 1_b_0 is bankrupt!
1024:  Bank 0_b_0 is bankrupt!
964:  Bank 1_b_0 is bankrupt!
1053:  Bank 0_b_0 is bankrupt!
970:  Bank 1_b_0 is bankrupt!
1025:  Bank 0_b_0 is bankrupt!


#### 1 sensitivity

In [42]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 1              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_7")

In [21]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_7")

With corrected sales value of firms selling in foreign markets:

In [43]:
experiment(rand_seeds[21:24], 1200, "exch_7_corrected")

738:  Bank 0_b_0 is bankrupt!
815:  Bank 1_b_0 is bankrupt!
774:  Bank 1_b_0 is bankrupt!
809:  Bank 0_b_0 is bankrupt!
779:  Bank 0_b_0 is bankrupt!
805:  Bank 1_b_0 is bankrupt!


#### 1.1 sensitivity  (not used)

In [16]:
model_tools.pegged_rates = [-1,-1]            # which currencies are pegged together, -1 if to none
model_tools.trade_sensitivity = 1.1              # sensitivity in exchange factors updating    ## gamma
model_tools.normal_variance = 0.002

In [ ]:
#experiment(rand_seeds[21:24], 1200, "exch_8")

In [23]:
for seeds in rand_seeds[21:24]:
    aa = Sim(seeds[0], seeds[1])
    aa.run(1200)                     # 100 years
    aa.finalize_data()
    aa.export_data(note = "exch_8")

With corrected sales value of firms selling in foreign markets:

In [17]:
experiment(rand_seeds[21:24], 1200, "exch_8_corrected")

747:  Bank 1_b_0 is bankrupt!
749:  Bank 0_b_0 is bankrupt!


C:\Users\am\model_tools.py:1544: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1544: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1530: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1777: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_sentiment', 'hsh_income', 'savings_T',
       'budget', 'cons', 'cons_value', 'forced_savings', 'hsh_deposits',
       'prices', 'offered_wages', 'production', 'production_value',
       'labor_demand', 'labor', 'job_offers', 'vacancies', 'inventories',
       'sales', 'sales_value', 'sales_costs',

120:  Bank 1_b_0 is bankrupt!


C:\Users\am\model_tools.py:1001: RuntimeWarning: overflow encountered in multiply
  self.exch_factors = (1 + tr_sen/self.total_product * balances + random_factors)*self.exch_factors
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\numeric.py:1203: RuntimeWarning: invalid value encountered in multiply
  return multiply(a.ravel()[:, newaxis], b.ravel()[newaxis, :], out)


OverflowError: cannot convert float infinity to integer

### more detail on 0.2 - 0.4

In [21]:
grid = [[0.24], [0.28], [0.32], [0.36]]

In [22]:
exp_grid(grid, ["trade_sensitivity"], rand_seeds[21:24], 1200, "exch_detailed", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished


#### addtional runs

In [32]:
grid2 = [[0.30], [0.32], [0.36]]

In [33]:
exp_grid(grid2, ["trade_sensitivity"], rand_seeds[24:28], 1200, "exch_detailed_add", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished


# Experiments with global bank

## 2 large countries bank experiments

In [ ]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()
model_tools.global_bank = True

### baseline bank experiments

In [21]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.global_bank = True  

In [28]:
experiment(rand_seeds[24:27], 1200, "glob_bank_base")

With corrected sales value of firms selling in foreign markets:

In [22]:
experiment(rand_seeds[24:27], 1200, "base_glob_bank_corrected")

134:  Bank 0_b_0 is bankrupt!
147:  Bank 0_b_0 is bankrupt!
127:  Bank 0_b_0 is bankrupt!


### flexible wages in the country which does not have a bank (perifery)

In [23]:
model_tools.wage_resist = [12, 8]             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.08]              # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.global_bank = True  

In [ ]:
experiment(rand_seeds[24:27], 1200, "glob_bank_alt_perif")

With corrected sales value of firms selling in foreign markets:

In [24]:
experiment(rand_seeds[24:27], 1200, "alt_perif_glob_bank_corrected")

120:  Bank 0_b_0 is bankrupt!
138:  Bank 0_b_0 is bankrupt!
149:  Bank 0_b_0 is bankrupt!


### flexible wages in the country with the bank (centre)

In [25]:
model_tools.wage_resist = [8, 12]             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08, 0.05]              # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.global_bank = True  

In [32]:
experiment(rand_seeds[24:27], 1200, "glob_bank_alt_centre")

With corrected sales value of firms selling in foreign markets:

In [26]:
experiment(rand_seeds[24:27], 1200, "alt_centre_glob_bank_corrected")

130:  Bank 0_b_0 is bankrupt!
157:  Bank 0_b_0 is bankrupt!
155:  Bank 0_b_0 is bankrupt!


### extremely flexible wages in the perifery

In [27]:
model_tools.wage_resist = [12, 8]             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05, 0.25]              # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.global_bank = True  

In [34]:
experiment(rand_seeds[24:27], 1200, "glob_bank_extr_alt_perif")

With corrected sales value of firms selling in foreign markets:

In [28]:
experiment(rand_seeds[24:27], 1200, "extr_alt_perif_glob_bank_corrected")

126:  Bank 0_b_0 is bankrupt!
147:  Bank 0_b_0 is bankrupt!
147:  Bank 0_b_0 is bankrupt!


### extremely flexible wages in the centre

In [29]:
model_tools.wage_resist = [8, 12]             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25, 0.05]              # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.global_bank = True  

In [36]:
experiment(rand_seeds[24:27], 1200, "glob_bank_extr_alt_centre")

With corrected sales value of firms selling in foreign markets:

In [30]:
experiment(rand_seeds[24:27], 1200, "extr_alt_centre_glob_bank_corrected")

131:  Bank 0_b_0 is bankrupt!
140:  Bank 0_b_0 is bankrupt!
149:  Bank 0_b_0 is bankrupt!


# 6 countries

In [27]:
model_tools.ns = [6000, 6000, 2000, 2000, 2000, 2000]    # no HSHs
model_tools.ms = [660, 660, 220, 220, 220, 220]      # no firms
set_base_pars()


## baseline 6C

In [23]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [10]:
experiment(rand_seeds[:4], 900, "base")   # 75 years

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
C:\Users\am\model_tools.py:1706: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1692: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))


With corrected sales value of firms selling in foreign markets:

In [28]:
%cd "d:\moje_dokumenty\Desktop\diplomka\model\data\hdf" 

d:\moje_dokumenty\Desktop\diplomka\model\data\hdf


In [29]:
experiment(rand_seeds[:4], 900, "base_corrected", csv = False)   # 75 years

C:\Users\am\model_tools.py:1544: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1530: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1544: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))


In [12]:
# finishing after pc shutdown
experiment(rand_seeds[3:4], 900, "base_corrected")   # 75 years

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
C:\Users\am\model_tools.py:1539: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1525: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1539: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1763: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_senti

## alternative 1 6C
1 large country has extremely flexible wages, others are in baseline

In [30]:
model_tools.wage_resist = [8] + [12]*(model_tools.no_countries-1)             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25] + [0.05]*(model_tools.no_countries-1)          # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [12]:
experiment(rand_seeds[:4], 900, "Extr_alt1")   # 75 years

With corrected sales value of firms selling in foreign markets:

In [31]:
experiment(rand_seeds[:4], 900, "Extr_alt1_corrected", csv = False)   # 75 years

736:  Bank 2_b_0 is bankrupt!
804:  Bank 5_b_0 is bankrupt!
670:  Bank 4_b_0 is bankrupt!
795:  Bank 3_b_0 is bankrupt!
808:  Bank 2_b_0 is bankrupt!
710:  Bank 2_b_0 is bankrupt!
723:  Bank 5_b_0 is bankrupt!
780:  Bank 4_b_0 is bankrupt!
896:  Bank 1_b_0 is bankrupt!
696:  Bank 5_b_0 is bankrupt!
755:  Bank 4_b_0 is bankrupt!


sim 1:

736:  Bank 2_b_0 is bankrupt!

804:  Bank 5_b_0 is bankrupt!


sim 2:

670:  Bank 4_b_0 is bankrupt!

795:  Bank 3_b_0 is bankrupt!

808:  Bank 2_b_0 is bankrupt!


sim 3:

710:  Bank 2_b_0 is bankrupt!

723:  Bank 5_b_0 is bankrupt!

780:  Bank 4_b_0 is bankrupt!

896:  Bank 1_b_0 is bankrupt!


sim 4:

696:  Bank 5_b_0 is bankrupt!

755:  Bank 4_b_0 is bankrupt!

## alternative 1b 6C
in order to draw comparison with Large-Small country model aletrnative 2b, what happens when 1 large country has extremely flexible wages, is allowed to export to and import from one small country , and others are under baseline conditions?

In [32]:
model_tools.wage_resist = [8] + [12]*(model_tools.no_countries-1)             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25] + [0.05]*(model_tools.no_countries-1)          # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

model_tools.free_market = False                 # all quotas are 1?
mm = np.array([[1]*model_tools.no_countries]*model_tools.no_countries)                                     
mm[0], mm.T[0] = [1,0,1,0,0,0], [1,0,1,0,0,0]     # country 0 trades only with country 2
model_tools.trade_quotas = [list(country) for country in mm] 

In [14]:
experiment(rand_seeds[:4], 1200, "Extr_alt1b")   # 100 years

With corrected sales value of firms selling in foreign markets:

In [33]:
experiment(rand_seeds[:4], 1200, "Extr_alt1b_corrected", csv = False)   # 100 years

(wrong simulations)

sim 1:

736:  Bank 2_b_0 is bankrupt!

804:  Bank 5_b_0 is bankrupt!

927:  Bank 1_b_0 is bankrupt!

930:  Bank 3_b_0 is bankrupt!

1053:  Bank 4_b_0 is bankrupt!


sim 2:

670:  Bank 4_b_0 is bankrupt!

795:  Bank 3_b_0 is bankrupt!

808:  Bank 2_b_0 is bankrupt!

961:  Bank 1_b_0 is bankrupt!

970:  Bank 5_b_0 is bankrupt!


sim 3:

710:  Bank 2_b_0 is bankrupt!

723:  Bank 5_b_0 is bankrupt!

780:  Bank 4_b_0 is bankrupt!

896:  Bank 1_b_0 is bankrupt!

926:  Bank 3_b_0 is bankrupt!


sim 4:

696:  Bank 5_b_0 is bankrupt!

755:  Bank 4_b_0 is bankrupt!

904:  Bank 1_b_0 is bankrupt!

930:  Bank 3_b_0 is bankrupt!

1000:  Bank 2_b_0 is bankrupt!

# 6 countries with 8000 households in large ones

In [34]:
model_tools.ns = [8000, 8000, 2000, 2000, 2000, 2000]    # no HSHs
model_tools.ms = [880, 880, 220, 220, 220, 220]      # no firms
set_base_pars()


## alternative 1b 6C
in order to draw comparison with Large-Small country model aletrnative 2b (which had 8000 - 2000 households), what happens when 1 large country has extremely flexible wages, is allowed to export to and import from one small country , and others are under baseline conditions?

In [35]:
model_tools.wage_resist = [8] + [12]*(model_tools.no_countries-1)             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.25] + [0.05]*(model_tools.no_countries-1)          # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

model_tools.free_market = False                 # all quotas are 1?
mm = np.array([[1]*model_tools.no_countries]*model_tools.no_countries)                                     
mm[0], mm.T[0] = [1,0,1,0,0,0], [1,0,1,0,0,0]     # country 0 trades only with country 2
model_tools.trade_quotas = [list(country) for country in mm] 

In [17]:
experiment(rand_seeds[:4], 1200, "Extr_alt1b_8")   # 100 years

With corrected sales value of firms selling in foreign markets:

In [36]:
experiment(rand_seeds[:4], 1200, "Extr_alt1b_8_corrected", csv = False)   # 100 years

# Experiments with trade quotas

## 2 countries BB models

In [26]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()


### BB baseline quotas experiments

In [27]:
model_tools.wage_resist = [12]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.05]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


#### 0 quotas (1 country model x2)

In [28]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [29]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_0", csv = False)   #  75 years - just to see if cycles emerge

#### 0.05 quotas

In [39]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.05]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [40]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_1", csv = False)   #  75 years - just to see if cycles emerge

#### 0.1 quotas

In [41]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.1]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [42]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_2", csv = False)   #  75 years - just to see if cycles emerge

#### 0.2 quotas

In [43]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.2]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [44]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_3", csv = False)   #  75 years - just to see if cycles emerge

#### 0.3 quotas

In [45]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.3]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [46]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_4", csv = False)   #  75 years - just to see if cycles emerge

#### 0.5 quotas

In [47]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.5]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [48]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_5", csv = False)   #  75 years - just to see if cycles emerge

#### 0.7 quotas

In [49]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.7]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [50]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_6", csv = False)   #  75 years - just to see if cycles emerge

#### 0.8 quotas

In [51]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.8]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [52]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_7", csv = False)   #  75 years - just to see if cycles emerge

#### 0.9 quotas

In [53]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.9]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [54]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_8", csv = False)   #  75 years - just to see if cycles emerge

#### 0.95 quotas

In [55]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.95]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [56]:
experiment(rand_seeds[7:10], 900, "quotas_bb_base_9", csv = False)   #  75 years - just to see if cycles emerge

### BB alternative quotas experiments

In [30]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


#### 0 quotas (1 country model x2)

In [31]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [32]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_0", csv = False)   #  75 years - just to see if cycles emerge

#### 0.05 quotas

In [58]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.05]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [59]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_1", csv = False)   #  75 years - just to see if cycles emerge

#### 0.1 quotas

In [60]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.1]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [61]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_2", csv = False)   #  75 years - just to see if cycles emerge

#### 0.2 quotas

In [62]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.2]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [63]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_3", csv = False)   #  75 years - just to see if cycles emerge

#### 0.3 quotas

In [64]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.3]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [65]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_4", csv = False)   #  75 years - just to see if cycles emerge

#### 0.5 quotas

In [66]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.5]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [67]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_5", csv = False)   #  75 years - just to see if cycles emerge

#### 0.7 quotas

In [68]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.7]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [69]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_6", csv = False)   #  75 years - just to see if cycles emerge

#### 0.8 quotas

In [70]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.8]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [71]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_7", csv = False)   #  75 years - just to see if cycles emerge

#### 0.9 quotas

In [72]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.9]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [73]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_8", csv = False)   #  75 years - just to see if cycles emerge

#### 0.95 quotas

In [74]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0.95]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas] 

In [75]:
experiment(rand_seeds[14:17], 900, "quotas_bb_alt_9", csv = False)   #  75 years - just to see if cycles emerge

## 2 countries BS models

In [33]:
model_tools.ns = [8000,2000]    # no HSHs
model_tools.ms = [880,220]      # no firms
set_base_pars()
model_tools.free_market = False                 # all quotas are 1?


### BS baseline quotas experiments

#### note:
from now updated model with corrected loans payback and doubtful loans is used

In [12]:
quotas = [0.05, 0.1, 0.2, 0.3, 0.5, 0.7, 0.8, 0.9, 0.95]

In [18]:
grid = [[make_matrix(quota, False)] for quota in quotas]

In [21]:
exp_grid( grid , ["trade_quotas"],  
         rand_seeds[14:17], 900, "base_quotas_bs", csv = False)

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
C:\Users\am\model_tools.py:1556: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1542: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1556: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1789: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_senti

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished


In [34]:
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas]

In [35]:
experiment(rand_seeds[14:17], 900, "base_quotas_bs_0", csv = False)  

### BS alternative quotas experiments

In [22]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative
model_tools.free_market = False 

In [23]:
quotas = [0.05, 0.1, 0.2, 0.3, 0.5, 0.7, 0.8, 0.9, 0.95]

In [24]:
grid = [[make_matrix(quota, False)] for quota in quotas]

In [25]:
exp_grid( grid , ["trade_quotas"],  
         rand_seeds[14:17], 900, "alt_quotas_bs", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished


In [36]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative

model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_quotas = np.array([[0]*model_tools.no_countries]*model_tools.no_countries)                                     
np.fill_diagonal(model_tools.trade_quotas, [1]*model_tools.no_countries) 
model_tools.trade_quotas = [list(country) for country in model_tools.trade_quotas]

In [37]:
experiment(rand_seeds[14:17], 900, "alt_quotas_bs_0", csv = False)  

# Trade costs experiments

## BB trade costs

In [9]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()


### baseline BB trade costs

In [24]:
costs = [0, 0.01, 0.1, 0.2, 0.3, 0.5, 0.7, 1, 1.3]     # 0.05 is the original value

In [25]:
grid = [[make_matrix(cost, True)] for cost in costs]

In [46]:
exp_grid( grid , ["trade_costs"],  
         rand_seeds[14:17], 900, "base_costs_bb", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished


### alternative BB costs

In [10]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [11]:
costs = [0, 0.01, 0.1, 0.2, 0.3, 0.5, 0.7, 1, 1.3]     # 0.05 is the original value
grid = [[make_matrix(cost, True)] for cost in costs]

In [ ]:
exp_grid( grid , ["trade_costs"],  
         rand_seeds[14:17], 900, "alt_costs_bb", csv = False)

experiment 0 is finished
experiment 1 is finished


In [13]:
exp_grid( grid , ["trade_costs"],    # finishing after restart
         rand_seeds[14:17], 900, "alt_costs_bb", start = 2, csv = False)

C:\Users\am\model_tools.py:1556: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1542: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1556: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1789: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_sentiment', 'hsh_income', 'savings_T',
       'budget', 'cons', 'cons_value', 'forced_savings', 'hsh_deposits',
       'prices', 'offered_wages', 'production', 'production_value',
       'labor_demand', 'labor', 'job_offers', 'vacancies', 'inventories',
       'sales', 'sales_value', 'sales_costs',

306:  Bank 1_b_0 is bankrupt!
309:  Bank 0_b_0 is bankrupt!
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished


alt_costs_bb_2 first or second run:

306:  Bank 1_b_0 is bankrupt!

309:  Bank 0_b_0 is bankrupt!

In [12]:
# redo after mistakingly deleting results
model_tools.trade_costs = grid[1][0]

In [13]:
experiment(rand_seeds[14:17], 900, "alt_costs_bb_2", csv = False)  #       # second experiment numbered as 2

C:\Users\am\model_tools.py:1556: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1542: RuntimeWarning: invalid value encountered in true_divide
  results.append(list(diff/s1))
C:\Users\am\model_tools.py:1556: RuntimeWarning: divide by zero encountered in true_divide
  results.append(list(multiply*s1/s2))
C:\Users\am\model_tools.py:1789: PerformanceWarning: 
your performance may suffer as PyTables will pickle object types that it cannot
map directly to c-types [inferred_type->mixed,key->block0_values] [items->Index(['employment', 'wages', 'working_wages', 'vol_unemployed', 'months_u',
       'unemployed_months_u', 'hsh_sentiment', 'hsh_income', 'savings_T',
       'budget', 'cons', 'cons_value', 'forced_savings', 'hsh_deposits',
       'prices', 'offered_wages', 'production', 'production_value',
       'labor_demand', 'labor', 'job_offers', 'vacancies', 'inventories',
       'sales', 'sales_value', 'sales_costs',

## BS trade costs

In [14]:
model_tools.ns = [8000,2000]    # no HSHs
model_tools.ms = [880,220]      # no firms
set_base_pars()


### baseline BS trade costs

In [15]:
costs = [0, 0.01, 0.1, 0.2, 0.3, 0.5, 0.7, 1, 1.3]     # 0.05 is the original value
grid = [[make_matrix(cost, True)] for cost in costs]

In [16]:
exp_grid( grid , ["trade_costs"],  
         rand_seeds[14:17], 900, "base_costs_bs", csv = False)

C:\Users\am\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
C:\Users\am\Anaconda3\lib\site-packages\numpy\core\_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
776:  Bank 1_b_0 is bankrupt!
831:  Bank 1_b_0 is bankrupt!
734:  Bank 1_b_0 is bankrupt!
experiment 3 is finished
467:  Bank 1_b_0 is bankrupt!
498:  Bank 1_b_0 is bankrupt!
471:  Bank 1_b_0 is bankrupt!
experiment 4 is finished
323:  Bank 1_b_0 is bankrupt!
321:  Bank 1_b_0 is bankrupt!
325:  Bank 1_b_0 is bankrupt!
experiment 5 is finished
240:  Bank 1_b_0 is bankrupt!
261:  Bank 1_b_0 is bankrupt!
264:  Bank 1_b_0 is bankrupt!
experiment 6 is finished
165:  Bank 1_b_0 is bankrupt!
171:  Bank 1_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
experiment 7 is finished
165:  Bank 1_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
experiment 8 is finished


bs baseline costs experiments:

experiment 3:

776:  Bank 1_b_0 is bankrupt!

831:  Bank 1_b_0 is bankrupt!

734:  Bank 1_b_0 is bankrupt!

experiment 4:

467:  Bank 1_b_0 is bankrupt!

498:  Bank 1_b_0 is bankrupt!

471:  Bank 1_b_0 is bankrupt!

experiment 5:

323:  Bank 1_b_0 is bankrupt!

321:  Bank 1_b_0 is bankrupt!

325:  Bank 1_b_0 is bankrupt!

experiment 6:

240:  Bank 1_b_0 is bankrupt!

261:  Bank 1_b_0 is bankrupt!

264:  Bank 1_b_0 is bankrupt!

experiment 7:

165:  Bank 1_b_0 is bankrupt!

171:  Bank 1_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

experiment 8:

165:  Bank 1_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!
 

### alternative BS costs

In [17]:
model_tools.wage_resist = [8]*model_tools.no_countries             # wage resistance in unemployment; 12 in baseline, 8 in alternative
model_tools.wage_flex = [0.08]*model_tools.no_countries             # wage flexibility;                0.05 in baseline, 0.08 or 0.25 in alternative


In [18]:
costs = [0, 0.01, 0.1, 0.2, 0.3, 0.5, 0.7, 1, 1.3]     # 0.05 is the original value
grid = [[make_matrix(cost, True)] for cost in costs]

In [19]:
exp_grid( grid , ["trade_costs"],  
         rand_seeds[14:17], 900, "alt_costs_bs", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
661:  Bank 1_b_0 is bankrupt!
624:  Bank 1_b_0 is bankrupt!
723:  Bank 1_b_0 is bankrupt!
experiment 3 is finished
438:  Bank 1_b_0 is bankrupt!
461:  Bank 1_b_0 is bankrupt!
441:  Bank 1_b_0 is bankrupt!
experiment 4 is finished
362:  Bank 1_b_0 is bankrupt!
427:  Bank 1_b_0 is bankrupt!
410:  Bank 1_b_0 is bankrupt!
experiment 5 is finished
120:  Bank 1_b_0 is bankrupt!
632:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
208:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
211:  Bank 0_b_0 is bankrupt!
experiment 6 is finished
120:  Bank 1_b_0 is bankrupt!
250:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
244:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
468:  Bank 0_b_0 is bankrupt!
experiment 7 is finished
120:  Bank 1_b_0 is bankrupt!
231:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
471:  Bank 0_b_0 is bankrupt!
120:  Bank 1_b_0 is bankrupt!
261:  Bank 0_b_0 is 

BS alternative trade costs experiments:

experiment 3:

661:  Bank 1_b_0 is bankrupt!

624:  Bank 1_b_0 is bankrupt!

723:  Bank 1_b_0 is bankrupt!

experiment 4:

438:  Bank 1_b_0 is bankrupt!

461:  Bank 1_b_0 is bankrupt!

441:  Bank 1_b_0 is bankrupt!

experiment 5:

362:  Bank 1_b_0 is bankrupt!

427:  Bank 1_b_0 is bankrupt!

410:  Bank 1_b_0 is bankrupt!

experiment 6:

120:  Bank 1_b_0 is bankrupt!

632:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

208:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

211:  Bank 0_b_0 is bankrupt!

experiment 7:

120:  Bank 1_b_0 is bankrupt!

250:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

244:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

468:  Bank 0_b_0 is bankrupt!

experiment 8:

120:  Bank 1_b_0 is bankrupt!

231:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

471:  Bank 0_b_0 is bankrupt!

120:  Bank 1_b_0 is bankrupt!

261:  Bank 0_b_0 is bankrupt!


# animal spirits experiments

##  animal spirits in BB and BS baselines

In [26]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()

In [27]:
grid = make_grid( ([(5000,5000), (550,550)], [(8000,2000), (880, 220)]), 
                 ([[0], [0.2], [0.4], [0.6], [0.75], [0.8], [0.9], [1]])  )   # 0.7 is original value

In [30]:
exp_grid( grid , ["ns", "ms", "p"],  
         rand_seeds[28:31], 1200, "spirits_base", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished
experiment 9 is finished
experiment 10 is finished
experiment 11 is finished
experiment 12 is finished
experiment 13 is finished
experiment 14 is finished
experiment 15 is finished


# (almost) perfect union with quotas - pegged rates, 0 trade costs

In [45]:
model_tools.ns = [5000,5000]    # no HSHs
model_tools.ms = [550,550]      # no firms
set_base_pars()
model_tools.free_market = False                 # all quotas are 1?
model_tools.trade_costs = make_matrix(0, True)
model_tools.pegged_rates = [-1,0]

In [10]:
quotas = [0.01, 0.03, 0.05, 0.1, 0.3, 0.5, 0.75, 1]

In [11]:
grid = make_grid( ([[5000,5000], [550,550]], [[8000,2000], [880,220]]), 
                  ([make_matrix(quota, False)] for quota in quotas) )      

In [48]:
exp_grid( grid , ["ns", "ms", "trade_quotas"],  
         rand_seeds[23:26], 900, "perfect_union", csv = False)

experiment 0 is finished
experiment 1 is finished
experiment 2 is finished
experiment 3 is finished
experiment 4 is finished
experiment 5 is finished
experiment 6 is finished
experiment 7 is finished
experiment 8 is finished
experiment 9 is finished
experiment 10 is finished
experiment 11 is finished
experiment 12 is finished
experiment 13 is finished
experiment 14 is finished
experiment 15 is finished
